crates/testing/src/spinning_task.rs

```diff
@@ -210,13 +220,15 @@ where
                             if let Some(node) = self.handles.write().await.get_mut(idx) {
                                 tracing::error!("Node {} shutting down", idx);
                                 node.handle.shut_down().await;
+                                // For restarted nodes generate the network on correct view
+                                let generated_network = (self.channel_generator)(node_id).await;
```

Comment:
why do we generate the network on `RestartDown` rather than `RestartUp`?

Reply:
it should be fine to do either way, the main thing is to ensure we shut down the old nodes network before we generate the new network